databricks/koalas/typedef/typehints.py

```diff
@@ -104,8 +109,9 @@ def as_spark_type(tpe) -> types.DataType:
     # TODO: Add "boolean" and "string" types.
     # ArrayType
     if tpe in (np.ndarray,):
-        # TODO: support other child types
         return types.ArrayType(types.StringType())
+    elif isinstance(tpe, _GenericAlias) and (tpe.__origin__ in (list, typing.List)):
```

Comment:
Shall we use `hasattr(tpe, "__origin__")` to use the same way as in `infer_return_type`?

Reply:
Btw, I don't think `list` can be a generic type?
In that case, should we handle `list` as the same as `np.ndarray`? WDYT @HyukjinKwon ?